### DISCUSSION TITLE
Certain CSS styles not being applied when deployed on Vercel

### DISCUSSION THREAD

**Discussion by yijiuntay** [vercel/next.js#52018] [Help] [answered]
### Summary

Dear fellow devs,

I have deployed a Next.js project on Vercel but noticed that some css styles are not being applied, primarily flex related styles.

For example,
I have this in my local dev (running 'npm run dev'):
![image](https://github.com/vercel/next.js/assets/86616844/6f770303-9a11-4f8a-821f-37cc6b02c0ce)

so essentially there is a display: flex and justify-content: space-between for the component
<img width="434" alt="image" src="https://github.com/vercel/next.js/assets/86616844/3f6c8d34-2d5e-4c82-b56a-ebd82c71a549">

but after deployed (or 'npm run build' and then 'npm run start'):
![image](https://github.com/vercel/next.js/assets/86616844/d5a7c795-f142-4bee-9769-8781754ad0bd)

the display: flex is missing when inspected:
![image](https://github.com/vercel/next.js/assets/86616844/71a0914f-6a19-42a7-975a-d8fc77ff60b2)

Wondering if anybody has encountered this before or knows the cause and how it can be solved. You may use this link https://github.com/yijiuntay/memory-game to clone the project and try to reproduce.

Thanks in advance ❤

### Example

https://github.com/yijiuntay/memory-game

**Comment by icyJoseph** [comment, marked as answer]
Seeing this when building locally too. Didn't copy your repo, but just the css bits that are affected. Strange... but i think I've seen this before, I'll give it a go at producing a minimal repro. Perhaps it is the nesting?

**Reply by icyJoseph** [reply]
@yijiuntay I think you have to refactor so that footerIteam is not directly under footer. Something like:

```css
.footer {
    display: flex;
    justify-content: center;
    gap: 2%;
  }

  & .footer .footerItem {
```

I guess the minification somehow process drops the flex rule... because it sees it up in `.footer`-

**Reply by icyJoseph** [reply]
Like if you remove the display flex on `.footer` it works fine for `.footerItem` -

**Reply by yijiuntay** [reply]
yea, after some trial and error, I think it's the nesting. I was able to fix the issue by rearranging the nesting, though I'm still not quite sure why the fix works

**Reply by icyJoseph** [reply]
I think it is the CSS minifier that's tripping itself. I'll do a bit of testing for that this weekend.

**Comment by Hanprogramer** [comment]
Same goes for me, `object-position: top;` is somehow removed automatically. I tried adding !important and not working as well

**Comment by caocanx** [comment]
Version 14.1.0 still have this issue，some nested `display: flex` auto removed after build. Some on can help? I just want use the normal nested css file.

**Reply by caocanx** [reply]
Add `!important` could make it work, but it just ugly.

**Comment by prokopious** [comment]
I am having an issue where my Next.js project's css is applied when I run it locally, but when deployed on Vercel, no styles at all are applied. I am not a frond-end specialist. This is just a portfolio for Python jobs, and it looks hideous without styles.

**Reply by caocanx** [reply]
You can try [sass](https://nextjs.org/docs/app/building-your-application/styling/sass), it work well for me.

**Reply by grantbrewster** [reply]
this happens to me too, I can't seem to figure it out and I'm going crazy

**Reply by danidg2588** [reply]
This is the answer, install and use sass, the same happened to me, and sass fixed it, I also think the problem is the nesting.

**Comment by danidg2588** [comment]
You may try using sass, it is pretty much likely to work for you.
